distributed/worker.py

```diff
@@ -3392,24 +3414,16 @@ def ensure_computing(self) -> None:
                 pdb.set_trace()
             raise
 
-    async def execute(self, key: str, *, stimulus_id: str) -> None:
+    async def execute(self, key: str, *, stimulus_id: str) -> StateMachineEvent | None:
         if self.status in {Status.closing, Status.closed, Status.closing_gracefully}:
-            return
-        if key not in self.tasks:
-            return
-        ts = self.tasks[key]
+            return None
+        ts = self.tasks.get(key)
+        if not ts:
+            return None
+        if ts.state == "cancelled":
+            return CancelComputeEvent(key=ts.key, stimulus_id=stimulus_id)
```

Comment:
I would suggest a name like `AlreadyCancelledEvent` to distinguish this from `Client.cancel` events passed through the scheduler. We don't have this, yet, but there is some potential for confusion

Reply:
I split them now